torch/_ops.py

```diff
@@ -25,6 +25,115 @@ def dl_open_guard():
+
+    def __call__(self, *args, **kwargs):
+        return self.__op(*args, **kwargs or {})
+
+    def __getattr__(self, key):
+        return getattr(self.__op, key)
+
+    # `my_namespace::my_op`
+    @property
+    def name(self):
+        return "{}.{}".format(*self.__schema.name.split("::"))
+
+    @property
+    def overload_name(self):
+        return self.__schema.overload_name
+
+    @property
+    def overloadpacket(self):
+        return self.__overloadpacket
+
+    @property
+    def op(self):
+        return self.__op
+
+    # TODO: add more methods to expose information about input and output arguments
+
+class OpOverloadPacket():
+    def __init__(self, qualified_op_name, op_name, op):
+        # These attributes are accessible on the object through the properties
+        # defined below but are immutable
+        self.__qualified_op_name = qualified_op_name
+        self.__op_name = op_name
+        self.__op = op
+
+    # it's a no-op since OpOverloadPacket object is immutable and must be for a given op.
+    def __deepcopy__(self, memo=None):
+        return self
+
+    def __str__(self):
+        return "OpOverloadPacket(op='{}.{}')".format(*self.__qualified_op_name.split("::"))
+
+    @property
+    def qualified_op_name(self):
+        return "{}.{}".format(*self.__qualified_op_name.split("::"))
+
+    @property
+    def op_name(self):
+        return self.__op_name
+
+    @property
+    def op(self):
+        return self.__op
+
+    def __getattr__(self, key):
+        # It is not a valid op_name when __file__ is passed in
+        if key == '__file__':
+            return 'torch.ops'
+
+        throw_error = False
+
+        try:
+            use_key = "" if key == 'default' else key
+            # TODO: disallow access to overloads registered by JIT
+            op_ = torch._C._get_operation_overload(self.__qualified_op_name, use_key)
+            schema = torch._C._get_schema(self.__qualified_op_name, use_key)
+            overload = OpOverload(self, op_, schema)
+            # cache the overload object
+            setattr(self, key, overload)
+            return overload
+        except RuntimeError:
+            try:
+                # This is added to maintain bc in case the user queries an attribute that exists on `self.__op`
+                # which used to be returned before instead of the OpOverloadPacket
+                out = getattr(self.__op, key)
+                return out
+            except AttributeError:
+                throw_error = True
+
+        # We throw an error here as opposed to above since the stack trace and error message
+        # printed in that case is a bit confusing:
+        # RuntimeError: Found no matching schema
+        # During handling of the above exception, another exception occurred:
+        # AttributeError ...
+        if throw_error:
+            raise AttributeError("'{}' object has no attribute '{}'".format(str(self), key))
```

Comment:
I think you can set the exception context to get rid of this logic. See more details here:
https://stackoverflow.com/a/52725410

Reply:
sweet! thanks for the suggestion -- that works